src/app/map/map.component.ts

```diff
@@ -278,7 +279,91 @@ export class MapComponent implements OnInit {
   }
 
   public onMouseClickFireHydroQueryFirePerimeter() { 
-    // Issue #58: see demo app for pointers
+    this.loader = true;
+    this.createMessage('Querying layers, please wait...');
+    Object.keys(this.workflowLayers).forEach(layerName => {
+      if (layerName === 'Active WildFire Perimeters' || layerName === 'Archived WildFire Perimeters') {
+        this.workflowLayers[layerName].query().nearby(this.clickPoint, 4).returnGeometry(true)
+          .run((error: any, results: any) => {
+            this.findFeatures(error,results,layerName);
+          }
+        );
+      } else if (layerName === 'MTBS Fire Boundaries') {
+        this.workflowLayers[layerName].identify().on(this._mapService.map).at(this.clickPoint).returnGeometry(true).tolerance(5)
+          .run(async (error: any, results: any) => {
+            this.findFeatures(error,results,layerName);
+          }
+        );
+      }
+    });
+  }
+
+  public async findFeatures(error,results,layerName) {
+    let count = 0;
+    let popupcontent;
+    let selectedPerimeters = [];
+    let layer;
+    const shownFields = ['INCIDENTNAME', 'COMMENTS', 'GISACRES', 'FIRE_YEAR', 'CREATEDATE', 'ACRES', 'AGENCY', 'SOURCE', 'INCIDENT', 'FIRE_ID', 'FIRE_NAME', 'YEAR', 'STARTMONTH', 'STARTDAY', 'FIRE_TYPE'];
+
+    if (error) {
+      this.createMessage('Error occurred, check console','error');
+      this.loader = false;
+    } 
+    if (results && results.features.length > 0) {
+      results.features.forEach(feat => {
+        popupcontent = '<div class="popup-header"><b>' + layerName + ':</b></div><br>';
+        if(layerName === 'MTBS Fire Boundaries'){
+          let date = feat.properties.STARTMONTH + '/' + feat.properties.STARTDAY + '/' + feat.properties.YEAR;
+          if (date.indexOf('undefined') > -1) date = 'N/A';
+        }
+        Object.keys(feat.properties).forEach(prop => {
+          if (shownFields.indexOf(prop.toUpperCase()) > -1) {
+            let val = feat.properties[prop];
+            if (prop.toLowerCase().indexOf('date') > -1) {
+              val = new Date(val).toLocaleDateString();
+            }
+            popupcontent += '<b>' + prop + ':</b> ' + val + '<br>';
+          }
+        });
+        popupcontent += '<br>';
+        if (layerName === 'MTBS Fire Boundaries'){
+          layer = L.geoJSON(feat.geometry);
+        } else if(layerName === 'Active WildFire Perimeters' || layerName === 'Archived WildFire Perimeters'){
+          const col = layerName.indexOf('Active') > -1 ? 'yellow' : 'red';
+          layer = L.geoJSON(feat.geometry, {style: {color: col}});
+        }
+        layer.addTo(this._mapService.map);
+        this.addBurnPoint(layer.getBounds().getCenter(), popupcontent);
```

Comment:
This adds a point at the centroid of the polygon, which can look strange depending on the polygon's geometry. Example: 
![image](https://user-images.githubusercontent.com/40237491/135313910-bacaf751-7340-4bf6-adf8-cc66e03894aa.png)

Recommend that you change the point to be the clickpoint (I think this is how the demo app is done).

Reply:
The demo app does the same thing I am doing. It does it this way because sometimes fire perimeters are on top of one another. This way if you click a point that includes more than one fire perimeter the popup markers are not all right on top of each other. 
![image](https://user-images.githubusercontent.com/59063181/135317683-b105f596-f3db-46cd-b412-1001bcfef885.png)

Reply:
Ah, sorry about that! Wish there was a better way, but that makes sense.